src/Mod/Sketcher/App/planegcs/GCS.cpp

```diff
@@ -1613,16 +1619,12 @@ void System::initSolution(Algorithm alg)
             return;
     }
     std::vector<Constraint *> clistR;
-    if (!redundant.empty()) {
-        for (std::vector<Constraint*>::const_iterator constr = clist.begin(); constr != clist.end();
-             ++constr) {
-            if (redundant.count(*constr) == 0)
-                    clistR.push_back(*constr);
+    for (auto constr = clist.cbegin(); constr != clist.cend(); ++constr) {
+        if (redundant.empty() || redundant.count(constr->get()) != 0)
```

Comment:
Compared to actual code, this is opposite logic. Is that wanted ?

Reply:
Done.